### PR TITLE
[bitnami/elasticsearch] Elasticsearch hpa

### LINKED ISSUE #6551: [bitnami/magento], [bitnami/redis] Add HorizontalPodAutoscaler to chart
Edit: added Redis to feature suggestion as well.

**Which chart**:
bitnami/magento, bitnami/redis, latest

**Is your feature request related to a problem? Please describe.**
It'd be good to be able to enable a HPA for Magento + Elasticsearch, and Redis and not having to create and manage one manually.

**Describe the solution you'd like**
Being able to enable autoscaling and set settings via helm chart values would be good. Something like https://github.com/kubernetes/ingress-nginx/blob/9e89951ec76045f69058a6bd68692715364627d5/charts/ingress-nginx/values.yaml#L318

**Describe alternatives you've considered**
Creating manually, but doing this means it is managed separately which is not ideal.

### PR DESCRIPTION
**Pull request by mahgo**
**Description of the change**
Added Horizontal Pod Autoscaler to the Elasticsearch chart

**Benefits**
 Being able to enable HPAs when using the chart.

**Applicable issues**

  - fixes #6551

**Additional information**
Let me know if I've missed anything, there was more to do in this chart compared to the other 2 PRs I did adding HPAs (Magento, Redis).

I also tidied up the readme parameters as it was previously one big table.

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/elasticsearch/templates/master-hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.master.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "elasticsearch.master.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: data
```
Comment (alemorcuq) [inline bitnami/elasticsearch/templates/master-hpa.yaml:8]:
```suggestion
    app.kubernetes.io/component: master
```

**Review by alemorcuq — CHANGES_REQUESTED** [review]
Thank you very much for the PR, @mahgo !

**Comment by mahgo** [comment]
Fixed.

**Review by alemorcuq — APPROVED** [review]
Thanks!

**Pull request merged**